projects/extension/src/containers/Popup.tsx

```diff
@@ -20,27 +28,61 @@ const SearchBar: React.FC = (): React.ReactElement => (
 );
 
 const Popup: React.FunctionComponent = () => {
+  const [apps, setApps] = React.useState<AppType[]>([{} as AppType]);
+  const [activeTab, setActiveTab] = React.useState<React.ReactElement | undefined>();
+  const [rTabs, setRTabs] = React.useState<React.ReactElement[]>([]);
   const appliedTheme = createMuiTheme(light);
-  const tabs = useTabs();
-  /* this is kinda of how  the access is to chrome tabs */
-  chrome.tabs.query({'active': true, 'windowId': chrome.windows.WINDOW_ID_CURRENT}, 
-  (tabs) => {
-    console.log('tabs', tabs, tabs[0].url)
-  });
-  const tabCurrent = tabs[0];
-  const tabsRest = tabs.slice(1);
+
+  React.useEffect((): void => {
+    const port = chrome.runtime.connect({ name: `substrateExtension` });
+    port.onMessage.addListener(( { type, about, payload }): void => {
+      if (type === 'error') {
+        console.error('ERROR: ', payload);
+      } else {
+        about === 'apps' && setApps(payload);
+      }
+    });
+  }, []);
+
+  React.useEffect((): void => {
+    const gatherTabs: TabInterface[] = [];
+    const restTabs: React.ReactElement[] = [];
+    chrome.tabs.query({"currentWindow": true, }, tabs => {
+      console.log('apps', apps, tabs);
+      tabs.forEach(t => {
+        apps?.find(({ tabId, smoldotName, appName, }) => {
+          if (tabId === t.id) {
+            gatherTabs.push({
+              isActive: t.active,
+              tabId: t.id,
+              url: t.url,
+              uApp: {
+                networks: [smoldotName],
+                name: appName,
+                enabled: true
+              }
+            })
+          }
+        })
+      });
+      gatherTabs.forEach(t => {
+        (t.isActive) ? setActiveTab(<Tab current tab={t} />) : restTabs.push(<Tab key={t.tabId} tab={t}/>);
+      })
+      setRTabs(restTabs);
+    });
+  }, [apps]);
 
 	return (
     <ThemeProvider theme={appliedTheme}>
       <Box width={'340px'} mb={0.1}>
         <GlobalFonts />
-        <Tab current tab={tabCurrent}/>
+        {activeTab}
         <SearchBar />
         <Box marginY={1}>
-          {tabsRest.map(tab => <Tab key={tab.uApp.name} tab={tab}/>)}
+          {rTabs.map(r => r)}
         </Box>
         <Divider />
-        <MenuButton fullWidth>My Networks</MenuButton>
+        <MenuButton fullWidth onClick={() => openInNewTab('chrome-extension://npfgeeonalpabakkpdobjngodgljemkb/options.html')}>My Networks</MenuButton>
```

Comment:
This link is not stable  .. I think the correct way to do this is `chrome.runtime.openOptionsPage()`